apps/meteor/app/livechat/server/lib/RoutingManager.ts

```diff
@@ -226,6 +227,37 @@ export const RoutingManager: Routing = {
 			}),
 		);
 
+		if (isSingleContactEnabled() && inquiry.v.contactId) {
+			const contact = await LivechatContacts.findOne({
+				_id: inquiry.v.contactId,
+			});
+
+			if (!contact) {
+				logger.debug(`Could not find associated contact with visitor ${inquiry.v._id}`);
+				return room;
+			}
+
+			if (contact.unknown && settings.get<boolean>('Livechat_Block_Unknown_Contacts')) {
+				return room;
+			}
+
+			const contactVerificationApp = settings.get('Livechat_Contact_Verification_App');
+			// Note: Non-empty `Livechat_Contact_Verification_App` means the user has a Contact Verification App setup,
+			//       therefore, we must give the app control over the room
+			if (contactVerificationApp !== '') {
+				// Note: If it is not `Livechat_Request_Verification_On_First_Contact_Only` it means that even though the contact
+				//       was already verified, we must verify it again in order to handle the livechat conversation down to the queue
+				const verifiedChannels = contact.channels?.filter((channel) => channel.verified && channel.name === room.source.type) || [];
+				if (
+					!settings.get<boolean>('Livechat_Request_Verification_On_First_Contact_Only') ||
+					(verifiedChannels.length === 0 && settings.get<boolean>('Livechat_Block_Unverified_Contacts'))
+				) {
+					await unverifyContactChannel(contact, room.source.type, inquiry.v._id);
+					return room;
+				}
+			}
+		}
+
 		logger.debug(`Attempting to take Inquiry ${inquiry._id} [Agent ${agent.agentId}] `);
 
 		const { _id, rid } = inquiry;
```

Comment:
Think we should do these checks before checking on the contact cause it doesn't make sense to check on a contact verification if the room is closed or is already served.